optimism/src/mips/interpreter.rs

```diff
@@ -270,7 +326,16 @@ pub fn interpret_itype<Env: InterpreterEnv>(env: &mut Env, instr: ITypeInstructi
         ITypeInstruction::AndImmediate => (),
         ITypeInstruction::OrImmediate => (),
         ITypeInstruction::XorImmediate => (),
-        ITypeInstruction::LoadUpperImmediate => (),
+        ITypeInstruction::LoadUpperImmediate => {
+            // lui $reg, [most significant 16 bits of immediate]
+            let rt = env.get_instruction_part(InstructionPart::RT);
+            let immediate_value = env.get_immediate();
```

Comment:
According to [this](https://www.dsi.unive.it/~gasparetto/materials/MIPS_Instruction_Set.pdf),

> 6 Load constant into upper 16 bits.
> Lower 16 bits are set to zero.

 shouldn't this multiply the immediate value by 2^16 and then overwrite the register?